test/reporters/json.js

```diff
@@ -12,7 +12,7 @@ describe('json reporter', function(){
     });
     suite = new Suite('JSON suite', 'root');
     runner = new Runner(suite);
-    var mochaReporter = new mocha._reporter(runner);
+    var mochaReporter = new mocha._reporters[0].fn(runner);
```

Comment:
Hopefully plugins/libraries don't break because we no longer expose `_reporter`.